python/py-terminaltables/Portfile

```diff
@@ -24,7 +25,17 @@ checksums           rmd160  9d02523f89d32c890ae9c40212155e221584e279 \
                     size    12264
 
 if {${name} ne ${subport}} {
-    depends_build-append    port:py${python.version}-setuptools
+    patch.pre_args  -p1
+    patchfiles      10-fix-poetry.patch
+
+    pre-patch {
+        # Convert files from DOS to UNIX linefeeds before patching.
+        reinplace "s/\r$//g" ${worksrcpath}/pyproject.toml
```

Comment:
this seems a very convoluted way of making the patch apply, why not just make a patchfile yourself that will apply cleanly without the need of changing linefeeds?

Reply:
I think the pep option in the PG fixes this, I'm doing some updates right now.

Reply:
Oh nevermind this was my pre-patch things, sorry

Reply:
Unless I am missing something, `patch` on pre-macOS 13 (aka darwin<22) is a modified version of `gpatch` where the `--binary` option is nullified by not setting `HAVE_SETMODE_DOS`

You can see this at Apple's releases:

https://github.com/apple-oss-distributions/gpatch/blob/gpatch-4/patch/patch.c#L586-L590

So this means any patch file with EOL of CRLF we feed into this will always get its CR characters stripped.

Using the `-l` flag to ignore whitespaces on darwin>=22 works because of gpatch being ditched.

I don't know if there's a better solution to this, but currently stripping the CRs from target seems to be the only solution.

Reply:
Turns out I _was_ missing something (sorry!), I created a local git repo using the terminaltables' python distribution tar.gz. And then made my change of `dos2unix`ing `pyproject.toml` and the `poetry` to `poetry-core` to the same file. After comitting, I created the patch file with `git-format-patch`/`git format-patch`.

Now this auto-created patch file works on all CI runners with the `-p1` pre_flag (this is due to the PG appends `-p0` if nothing is set on pre_flag).

I don't know _what_ `git-format-patch` or I did; but during this patching rabbit hole, I created a CI environment similar to this repo which helped me _a ton_.

I'm so sorry for all the bothering 😞